pkg/networkserver/downlink.go

```diff
@@ -881,14 +882,15 @@ func (ns *NetworkServer) scheduleDownlinkByPaths(ctx context.Context, req *sched
 		}
 
 		var a *attempt
-		if len(attempts) > 0 {
-			if last := attempts[len(attempts)-1]; last.Equal(target) {
-				a = last
+		for _, attempt := range attempts {
```

Comment:
Previously we would merge these attempts only if they were equal to the last target. But this doesn't really make much sense to me, since it would result in the NS doing 3 GS RPCs instead of 2 for the following set of paths:
- Gateway 1 on peer 1.
- Gateway 2 on peer 2.
- Gateway 3 on peer 1.

I understand that the previous implementation was linear with respect to the number of paths, but I think things got prematurely optimized.

Now gateway 1 and 3 are merged on the same attempt. As a result of this change the downlinks tests have been updated to reflect this merging behavior.

Reply:
You're entirely correct that this change breaks the ordering of the paths - my bad. I've fixed this in 7cf7e5b6b5e59ea041106448dcf13987997e873f.

With respect to how much any round trip optimizations are worth - in P95 we actually can see 20ms at times, due to cross-availability-zone communication.